lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -167,43 +223,46 @@ def self.is_firebasecmd_supported?(cmd)
         true
       end
 
-      def self.get_token
-        client = Signet::OAuth2::Client.new(
-          token_credential_uri: 'https://oauth2.googleapis.com/token',
-          client_id: FirebaseAppDistributionLoginAction::CLIENT_ID,
-          client_secret: FirebaseAppDistributionLoginAction::CLIENT_SECRET,
-          refresh_token: ENV["FIREBASE_TOKEN"]
-        )
-
-        client.fetch_access_token!
-        return client.access_token
-      rescue Signet::AuthorizationError => error
-        UI.crash!("Wrong value for FIREBASE_TOKEN")
-      end
-
-      def self.validate_app!(app_id)
-        token = get_token
-
-        connection = Faraday.new(url: BASE_URL) do |conn|
-          conn.response(:json, parser_options: { symbolize_names: true })
-          conn.response(:raise_error) # raise_error middleware will run before the json middleware
-          conn.adapter(Faraday.default_adapter)
-        end
+      def self.get_upload_token(app_id, binary_path)
+        binary_hash = Digest::SHA256.hexdigest(File.open(binary_path).read)
         begin
           response = connection.get("#{PATH}#{app_id}") do |request|
-            request.headers["Authorization"] = "Bearer " + token
+            request.headers["Authorization"] = "Bearer " + auth_token
           end
-        rescue Faraday::ResourceNotFound => error
-          UI.user_error!("App Distribution could not find your app #{app_id}. Make sure to onboard your app by pressing the \"Get started\" button on the App Distribution page in the Firebase console: https://console.firebase.google.com/project/_/appdistribution")
-        rescue => error
-          UI.crash!("Failed to fetch app information: #{error.message}")
+        rescue Faraday::ResourceNotFound
+          UI.user_error!(FirebaseAppDistributionError::MISSING_APP_ID)
+        rescue
+          UI.crash!(FirebaseAppDistributionError::GET_APP_ERROR)
```

Comment:
This rescue block can be called for a number of reasons (permissions error, server error, throttling etc.). I think it would be better for debugging to pass the exception to the `UI.crash` method instead of passing a generic message i.e.

```
rescue => error
  puts UI.crash!(error)
end
```